Source/WebKit/Configurations/DerivedSources.xcconfig

```diff
@@ -0,0 +1,34 @@
+// Copyright (C) 2022 Apple Inc. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+// 1. Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+// 2. Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+
+#include "BaseTarget.xcconfig"
+
+INSTALLHDRS_SCRIPT_PHASE = YES;
+
+// WebKit's private module map needs to be unifdef'ed, so the standard MODULEMAP_PRIVATE_FILE
+// can't be used. This target does the unifdef and WebKit's "Copy Private Module Map" Copy
+// Files build phase copies them into the framework.
+WK_WEBKIT_MODULEMAP_PRIVATE_FILE = $(WK_WEBKIT_MODULEMAP_PRIVATE_FILE_$(WK_COCOA_TOUCH)_$(WK_IS_CATALYST));
+WK_WEBKIT_MODULEMAP_PRIVATE_FILE_cocoatouch_NO = Modules/iOS_Private.modulemap;
+WK_WEBKIT_MODULEMAP_PRIVATE_FILE_cocoatouch_YES = Modules/MacCatalyst_Private.modulemap;
+WK_WEBKIT_MODULEMAP_PRIVATE_FILE__NO = Modules/OSX_Private.modulemap;
```

Comment:
So in other projects (like WebCore), instead of creating a separate set of configs for the Derived Sources target, we use the main framework's xcconfig to get settings like INSTALLHDRS_SCRIPT_PHASE. Would you mind doing that here as well? It would also mean going back to having the modulemap input path logic computed in the Makefile.

Here's the diff to change WebKit's Derived Sources to be based off of WebKit.xcconfig:
```diff
diff --git a/Source/WebKit/WebKit.xcodeproj/project.pbxproj b/Source/WebKit/WebKit.xcodeproj/project.pbxproj
index 122fc59c6720..748aa6389e08 100644
--- a/Source/WebKit/WebKit.xcodeproj/project.pbxproj
+++ b/Source/WebKit/WebKit.xcodeproj/project.pbxproj
@@ -18987,7 +18987,7 @@
                };
                C0CE72861247E66800BC0EC4 /* Debug */ = {
                        isa = XCBuildConfiguration;
-                       baseConfigurationReference = 5DAD73F1116FF90C00EE5396 /* BaseTarget.xcconfig */;
+                       baseConfigurationReference = BCB86F4B116AAACD00CE20B7 /* WebKit.xcconfig */;
                        buildSettings = {
                                COPY_PHASE_STRIP = NO;
                                GCC_DYNAMIC_NO_PIC = NO;
@@ -18998,7 +18998,7 @@
                };
                C0CE72871247E66800BC0EC4 /* Release */ = {
                        isa = XCBuildConfiguration;
-                       baseConfigurationReference = 5DAD73F1116FF90C00EE5396 /* BaseTarget.xcconfig */;
+                       baseConfigurationReference = BCB86F4B116AAACD00CE20B7 /* WebKit.xcconfig */;
                        buildSettings = {
                                COPY_PHASE_STRIP = YES;
                                DEBUG_INFORMATION_FORMAT = "dwarf-with-dsym";
@@ -19008,7 +19008,7 @@
                };
                C0CE72881247E66800BC0EC4 /* Production */ = {
                        isa = XCBuildConfiguration;
-                       baseConfigurationReference = 5DAD73F1116FF90C00EE5396 /* BaseTarget.xcconfig */;
+                       baseConfigurationReference = BCB86F4B116AAACD00CE20B7 /* WebKit.xcconfig */;
                        buildSettings = {
                                PRODUCT_NAME = "Derived Sources";
                        };
```

Reply:
If there's a good reason to have a separate DerivedSources.xcconfig that's escaping me, feel free to go ahead with the merge :)

Reply:
Sure we can use the same xcconfig. We could still make a build setting for the private module map so that the xcfilelist can just have the platform appropriate module map. That way it won’t rebuild e.g. iOS if the macOS module map changes

Reply:
> That way it won’t rebuild e.g. iOS if the macOS module map changes

Well, iirc it will rerun Make regardless becuase DerivedSources-input.xcconfig is platform-agnostic. But Make will avoid regenerating the private module map based on its own dependency checking.

Reply:
That’s true. Mostly I just like the module map files being declared next to each other in the xcconfig. It’ll also avoid the Run Script that runs make just for a no-op, but that’s really not a big deal like you say. However you prefer 😊

Reply:
> Mostly I just like the module map files being declared next to each other in the xcconfig.

That makes sense, putting the declarations side by side works for me!

Reply:
Ok, I think I got it this time. Also I noticed that the Mac Catalyst module map is the same as the iOS one so I added one more conditional and combined them. Can I get one more review please?

- deleted MacCatalyst_Private.modulemap
- deleted DerivedSources.xcconfig and switched that target to using the WebKit config
- moved the private module map settings to WebKit.xcconfig
- added WK_IS_CATALYST_FLAG to DerivedSources.make